core/src/main/scala/org/apache/spark/rdd/MapPartitionsWithPreparationRDD.scala

```diff
@@ -24,6 +24,8 @@ import org.apache.spark.{Partition, Partitioner, TaskContext}
 /**
  * An RDD that applies a user provided function to every partition of the parent RDD, and
  * additionally allows the user to prepare each partition before computing the parent partition.
+ *
+ * TODO(davies): remove this once SPARK-10342 is fixed
```

Comment:
Based on this comment, it is not clear what need to be removed (remove this class or remove changes of this PR). Can you add a comment in SPARK-10342 about what need to be removed?